### PR TITLE
miscellaneous cleanups and unifications

### PR DESCRIPTION
**Pull request by mmoll**
- use operating systems names without versions in the header
- reduce diff between kickstart RHEL and non-RHEL
- reduce diff between autoyast SLES and non-SLES
- don't use '-e 0' to suppress errors with yum

### CODE REVIEW

**Comment by sathieu** [comment]
looks good to me

File: kickstart/finish.erb [hunk 1]
```diff
@@ -2,15 +2,8 @@
 kind: finish
 name: Kickstart default finish
 oses:
```
Comment (domcleal) [inline kickstart/finish.erb:4]: missing "RedHat"?

Comment (mmoll) [inline kickstart/finish.erb:4]: there was no RHEL of any version associated before and then the `redhat_register` snippet should also go into there, I presume?

File: kickstart/userdata.erb [hunk 2]
```diff
@@ -2,15 +2,8 @@
 kind: user_data
 name: Kickstart default user data
 oses:
```
Comment (domcleal) [inline kickstart/userdata.erb:4]: missing RedHat?

Comment (mmoll) [inline kickstart/userdata.erb:4]: same comment as above

File: autoyast/provision_sles.erb [hunk 3]
```diff
@@ -124,6 +124,7 @@ cp /etc/resolv.conf /mnt/etc
 <% if salt_enabled %>
 <%= snippet 'saltstack_setup' %>
 <% end -%>
+
 <% if spacewalk_enabled -%>
```
Comment (domcleal) [inline autoyast/provision_sles.erb:128]: provision_rhel has this higher so it's able to install the packages if using Spacewalk etc.

File: autoyast/provision.erb [hunk 4]
```diff
@@ -113,6 +115,9 @@ cp /etc/resolv.conf /mnt/etc
         <notification>Setting up Puppet / Foreman ...</notification>
         <source><![CDATA[
 /bin/hostname <%= @host.name %>
+
+<%= snippet('remote_execution_ssh_keys') %>
+
 <% if puppet_enabled -%>
```
Comment (domcleal) [inline autoyast/provision.erb:121]: missing chef_client?

**Comment by domcleal** [comment]
The two {kickstart,preseed}/userdata.erb files could do with a slight refresh too, they're missing Chef, Salt and remote execution keys.

File: kickstart/finish.erb (hunk 1 above)
Comment (domcleal) [inline kickstart/finish.erb:4]: Hm, yes, it would be needed.  I don't mind if you want to leave it off.

File: autoyast/provision_sles.erb (hunk 3 above)
Comment (mmoll) [inline autoyast/provision_sles.erb:128]: For SLES that's only to register the system, main reason being that we add puppet/salt repos and packages in the first install step in the XML anyway. I do get the point, but if this needs to be changed that would probably require refactoring which can only be done when having such a setup available for testing.

File: autoyast/provision.erb (hunk 4 above)
Comment (mmoll) [inline autoyast/provision.erb:121]: We don't support chef on SUSE at the moment (I also did no research if and how that would be possible)

**Comment by mmoll** [comment]
tushed up the userdata files and added `('built')` for `foreman_url` in a few places.

**Comment by domcleal** [comment]
Merged, thanks @mmoll!

**Pull request closed without merging**